test/src/main/java/io/strimzi/test/k8s/Kubernetes.java

```diff
@@ -0,0 +1,314 @@
+        return CLIENT.inNamespace(namespace).apps().statefulSets().withName(statefulSetName).get().getSpec().getSelector();
+    }
+
+    /**
+     * Gets stateful set status
+     */
+    public boolean getStatefulSetStatus(String statefulSetName) {
+        return CLIENT.inNamespace(namespace).apps().statefulSets().withName(statefulSetName).isReady();
+    }
+
+    public void deleteStatefulSet (String statefulSetName) {
+        CLIENT.inNamespace(namespace).apps().statefulSets().withName(statefulSetName).delete();
+    }
+
+    public Deployment createOrReplaceDeployment (Deployment deployment) {
+        return CLIENT.inNamespace(namespace).apps().deployments().createOrReplace(deployment);
+    }
+
+    /**
+     * Gets deployment
+     */
+    public Deployment getDeployment(String deploymentName) {
+        return CLIENT.inNamespace(namespace).apps().deployments().withName(deploymentName).get();
+    }
+
+    /**
+     * Gets deployment status
+     */
+    public LabelSelector getDeploymentSelectors(String deploymentName) {
+        return CLIENT.inNamespace(namespace).apps().deployments().withName(deploymentName).get().getSpec().getSelector();
+    }
+
+    /**
+     * Gets deployment status
+     */
+    public boolean getDeploymentStatus(String deploymentName) {
+        return CLIENT.inNamespace(namespace).apps().deployments().withName(deploymentName).isReady();
+    }
+
+    public void deleteDeployment (String deploymentName) {
+        CLIENT.inNamespace(namespace).apps().deployments().withName(deploymentName).delete();
+    }
+
+    /**
+     * Gets deployment config status
+     */
+    public boolean getDeploymentConfigStatus(String deploymentCofigName) {
+        return CLIENT.inNamespace(namespace).adapt(OpenShiftClient.class).deploymentConfigs().withName(deploymentCofigName).isReady();
+    }
+
+    public Secret createSecret(Secret secret) {
+        return CLIENT.inNamespace(namespace).secrets().create(secret);
+    }
+
+    public Secret patchSecret(String secretName, Secret secret) {
+        return CLIENT.secrets().inNamespace(namespace).withName(secretName).patch(secret);
+    }
+
+
+    public Secret getSecret(String secretName) {
+        return CLIENT.inNamespace(namespace).secrets().withName(secretName).get();
+    }
+
+    public List<Secret> listSecrets() {
+        return CLIENT.inNamespace(namespace).secrets().list().getItems();
+    }
+
+    public Service getService (String serviceName) {
+        return CLIENT.inNamespace(namespace).services().withName(serviceName).get();
+    }
+
+    public boolean getServiceStatus (String serviceName) {
+        return CLIENT.inNamespace(namespace).services().withName(serviceName).isReady();
+    }
+
+    public void deleteService (String serviceName) {
+        CLIENT.inNamespace(namespace).services().withName(serviceName).delete();
+    }
+
+    public Job createJob(Job job) {
+        return CLIENT.inNamespace(namespace).extensions().jobs().create(job);
```

Comment:
`jobs()` in this context is deprecated, maybe find another solution which will be supported in future versions?